This addresses bug #599092

https://bugs.gentoo.org/show_bug.cgi?id=599092

commit 9ea3a6e150dfc822ba1565f649b634848597d2d9
Author: Daniel Stenberg <daniel@haxx.se>
Date:   Fri Nov 4 11:13:16 2016 +0100

    mbedtls: fix build with mbedtls versions < 2.4.0
    
    Regression added in 62a8095e714
    
    Reported-by: Tony Kelman
    
    Discussed in #1087

diff --git a/lib/vtls/mbedtls.c b/lib/vtls/mbedtls.c
index 24249dd..9f29ff0 100644
--- a/lib/vtls/mbedtls.c
+++ b/lib/vtls/mbedtls.c
@@ -31,11 +31,15 @@
 
 #ifdef USE_MBEDTLS
 
+#include <mbedtls/version.h>
+#if MBEDTLS_VERSION_NUMBER >= 0x02040000
 #include <mbedtls/net_sockets.h>
+#else
+#include <mbedtls/net.h>
+#endif
 #include <mbedtls/ssl.h>
 #include <mbedtls/certs.h>
 #include <mbedtls/x509.h>
-#include <mbedtls/version.h>
 
 #include <mbedtls/error.h>
 #include <mbedtls/entropy.h>
